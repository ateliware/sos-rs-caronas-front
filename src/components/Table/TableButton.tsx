import { PropsWithChildren } from 'react';

type Props = {
  active?: boolean;
  onClick?: () => void;
};

export default function TableButton(props: PropsWithChildren<Props>) {
  const buttonClasses = [
    'btn',
    'btn--icon-square',
    'btn--table-action',
    'ml-s-100',
  ];
  if (props.active) buttonClasses.push('btn--table-action--active');

  return (
    <button onClick={props.onClick} className={buttonClasses.join(' ')}>
      <span className="btn__body">
        <span className="btn__body__icon material-symbols-rounded">
          {props.children}
        </span>
      </span>
    </button>
  );
}
