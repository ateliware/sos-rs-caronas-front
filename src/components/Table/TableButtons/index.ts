export { default as FilterButton } from './FilterButton';
export { default as SortButton } from './SortButton';
