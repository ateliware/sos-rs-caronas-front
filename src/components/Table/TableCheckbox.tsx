type Props = {
  header?: boolean;
  checked?: boolean;
  onChange?: (checked: boolean) => void;
  onAction?: () => void;
};

function ActionButton(props: Pick<Props, 'onAction'>) {
  if (!props.onAction) return null;
  return (
    <button
      onClick={props.onAction}
      className="btn btn--icon-square btn--table-action ml-s-50"
    >
      <div className="btn__body">
        <span className="material-symbols-outlined btn__body__icon">
          arrow_drop_down
        </span>
      </div>
    </button>
  );
}

export default function TableCheckbox(props: Props) {
  if (props.header) {
    return (
      <th className="table__cell table__cell--checkbox">
        <div className="table__cell__content">
          <div className="form-check">
            <input
              checked={props.checked}
              onChange={(event) => props.onChange?.(event.target.checked)}
              className="form-check-input"
              type="checkbox"
              id="checkbox"
            />
          </div>
          <ActionButton onAction={props.onAction} />
        </div>
      </th>
    );
  } else {
    return (
      <td className="table__cell table__cell--checkbox">
        <div className="table__cell__content">
          <div className="form-check">
            <input
              checked={props.checked}
              onChange={(event) => props.onChange?.(event.target.checked)}
              className="form-check-input"
              type="checkbox"
              id="checkbox"
            />
          </div>
        </div>
      </td>
    );
  }
}
