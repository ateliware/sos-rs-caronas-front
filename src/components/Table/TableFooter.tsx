import { pageRange } from './TableHelper';

interface Props {
  currentPage: number;
  pageCount: number;
  setCurrentPage: React.Dispatch<React.SetStateAction<any>>;
  onPageChange: (page: number) => void;
}

export const TableFooter = (props: Props) => {
  return (
    <tfoot>
      <tr>
        <td>
          <nav className="pagination-container">
            <button
              disabled={props.currentPage === 1}
              onClick={() => {
                props.setCurrentPage(1);
                props.onPageChange!(1);
              }}
              className="pagination-button first"
              id="first-button"
              aria-label="first page"
              title="first page"
            >
              Primeira
            </button>
            <button
              disabled={props.currentPage === 1}
              onClick={() => {
                props.setCurrentPage((p: number) => p - 1);
                props.onPageChange!(props.currentPage - 1);
              }}
              className="pagination-button"
              id="prev-button"
              aria-label="Previous page"
              title="Previous page"
            >
              &lt;
            </button>

            <div id="pagination-numbers">
              {pageRange(props.currentPage, props.pageCount).map((i) => {
                const buttonClassNames = ['pagination-number'];

                if (i === props.currentPage) buttonClassNames.push('active');

                return (
                  <button
                    onClick={() => {
                      props.setCurrentPage(i);
                      props.onPageChange!(i);
                    }}
                    key={i}
                    className={buttonClassNames.join(' ')}
                  >
                    {i}
                  </button>
                );
              })}
            </div>

            <button
              disabled={props.currentPage === props.pageCount}
              onClick={() => {
                props.setCurrentPage((p: number) => p + 1);
                props.onPageChange!(props.currentPage + 1);
              }}
              className="pagination-button"
              id="next-button"
              aria-label="Next page"
              title="Next page"
            >
              &gt;
            </button>
            <button
              disabled={props.currentPage === props.pageCount}
              onClick={() => {
                props.setCurrentPage(props.pageCount);
                props.onPageChange!(props.pageCount);
              }}
              className="pagination-button last"
              id="last-button"
              aria-label="last page"
              title="Last page"
            >
              Ãšltima
            </button>
          </nav>
        </td>
      </tr>
    </tfoot>
  );
};
