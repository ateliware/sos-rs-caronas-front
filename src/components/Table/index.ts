export type { SelectedKeys } from './Table';
export type { ColumnProps } from './TableColumn';
export { default as Table } from './Table';
export { default as TableColumn } from './TableColumn';
