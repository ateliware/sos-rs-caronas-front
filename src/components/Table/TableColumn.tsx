import { ReactNode } from 'react';

import TableButton from './TableButton';
import { Filters } from './Table';
import { FilterButton, SortButton } from './TableButtons';

type CellProps = {
  container?: boolean;
  ellipsis?: boolean;
  cellAlign?: 'start' | 'end' | 'center' | 'between';
  verticalAlign?: 'top' | 'middle' | 'bottom';
  textAlign?: 'left' | 'center' | 'right';
  render?: (value: any, rowData: any) => ReactNode;
};

export type ColumnProps = {
  fromKey: string;
  header: ReactNode;
  headerAlign?: 'start' | 'end' | 'center' | 'between';
  disableSorting?: boolean;
  filterType?: 'text' | 'number' | 'date' | 'select';
  filterOptions?: { label: string; value: string | boolean | number }[];
  onAction?: (key: string) => void;
  setFilters?: React.Dispatch<React.SetStateAction<Filters>>;
  filters?: Filters;
  width?: string;
  className?: string;
} & CellProps;

export default function TableColumn(props: ColumnProps) {
  const key = props.fromKey;
  const thClassName = ['table__cell'];
  const thCellContentClassName = ['table__cell__content'];

  if (props.className) {
    thClassName.push(props.className);
  }

  if (props.headerAlign) {
    thCellContentClassName.push(`justify-${props.headerAlign}`);
    thCellContentClassName.push(`text-${props.headerAlign}`);
  }

  return (
    <th style={{ width: props.width }} className={thClassName.join(' ')}>
      <div className={thCellContentClassName.join(' ')}>
        {props.header}
        {!props.disableSorting && <SortButton {...props} />}
        {props.filterType && <FilterButton {...props} />}
        {props.onAction && (
          <TableButton onClick={() => props.onAction?.(key)}>
            more_horiz
          </TableButton>
        )}
      </div>
    </th>
  );
}
