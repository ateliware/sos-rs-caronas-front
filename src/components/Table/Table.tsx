import React, { Fragment, useId, useState } from 'react';
import { useForm } from 'react-hook-form';
import Skeleton from 'react-loading-skeleton';
import 'react-loading-skeleton/dist/skeleton.css';

import { RepositoryParams } from '@services/api';
import { debounce } from 'lodash';

import { Input, InputAddon, InputIcon } from '@components';

import TableCheckbox from './TableCheckbox';
import TableColumn, { ColumnProps } from './TableColumn';
import { TableFooter } from './TableFooter';
import TableRow from './TableRow';
import TableShowColumnSelector from './TableShowColumnSelector';

type Row = { [key: string]: any };
export type Filters = { [key: string]: any } & Partial<RepositoryParams>;

type Props = {
  className?: string;
  dataKey?: string;
  data: Row[];
  checkable?: boolean;
  hoverable?: boolean;
  layout?: 'fixed' | 'auto';
  rowPadding?: 'small' | 'medium' | 'large';
  children?: React.ReactElement<ColumnProps>[];
  selectedKeys?: SelectedKeys;
  invertSelection?: boolean;
  expandedRow?: string;
  isLoading?: boolean;
  withSearchInput?: boolean;
  withPagination?: boolean;
  filters?: Filters;
  setDragging?: React.Dispatch<React.SetStateAction<any>>;
  setFilters?: React.Dispatch<React.SetStateAction<Filters>>;
  onDrag?: (newIndex: number) => void;
  onDragEnd?: () => void;
  renderExpandedRow?: (row: any) => React.ReactNode;
  onAction?: (keys: SelectedKeys, invertSelection: boolean) => void;
  onSelectionChange?: (keys: SelectedKeys, invertSelection: boolean) => void;
  onRowClick?: (row: Row) => void;
  totalRecords?: number;
  showColumnSelector?: boolean;
};

export type SelectedKeys = { [key: string]: boolean };

export default function Table({ showColumnSelector = true, ...props }: Props) {
  const totalRecords = props.totalRecords || 0;

  const id = useId();
  const dataKey = props.dataKey ?? 'id';
  const tableClasses = ['table'];
  if (props.layout === 'fixed') tableClasses.push('table--fixed');
  if (props.hoverable) tableClasses.push('table--hoverable');
  if (props.className) tableClasses.push(props.className);
  if (props.rowPadding) tableClasses.push(`table--py-${props.rowPadding}`);
  const columns = props.children?.map((child) => child.props) || [];
  const [selectedKeys, setSelectedKeys] = useState<SelectedKeys>(
    props.selectedKeys ?? {}
  );
  const [invertSelection, setInvertSelect] = useState(
    props.invertSelection ?? false
  );
  const [currentPage, setCurrentPage] = useState(1);

  const paginationLimit = 10;
  const pageCount = Math.ceil(totalRecords / paginationLimit);

  const listFilter = debounce((query) => {
    return props.setFilters?.({ ...props.filters, q: query });
  }, 500);

  const { watch, register, setValue } = useForm();

  const data = props.data ?? [];
  return (
    <>
      <div className="table__filter col-md-12">
        <div className="d-flex justify-between">
          {props.withSearchInput && (
            <div className="col-md-10">
              <Input
                className="w-50"
                type="search"
                placeholder="Pesquisar"
                form={register('search', {
                  onChange: (e) => {
                    listFilter(e.target.value);
                  },
                })}
                prefixes={
                  <InputAddon left>
                    <InputIcon>search</InputIcon>
                  </InputAddon>
                }
              />
            </div>
          )}

          {showColumnSelector && (
            <TableShowColumnSelector
              watch={watch}
              setValue={setValue}
              register={register}
              columns={columns}
            />
          )}
        </div>
      </div>

      <table id={id} className={tableClasses.join(' ')}>
        <thead className="table__head">
          <tr className="table__row">
            {props.onDragEnd && (
              <TableColumn
                width="1px"
                headerAlign="center"
                fromKey="id"
                header="Ordem"
              />
            )}
            {props.checkable && (
              <TableCheckbox
                header
                onChange={() => {
                  setSelectedKeys({});
                  setInvertSelect(!invertSelection);
                  props.onSelectionChange?.({}, !invertSelection);
                }}
                onAction={
                  props.onAction &&
                  (() => props.onAction?.(selectedKeys, invertSelection))
                }
              />
            )}
            {React.Children.map(props.children, (child) => {
              const columnName = child!.props.header?.toString() || '';

              if (
                !showColumnSelector ||
                (showColumnSelector && watch(columnName, { value: true }))
              ) {
                if (React.isValidElement(child)) {
                  return React.cloneElement(child, {
                    filters: props.filters,
                    setFilters: props.setFilters,
                  });
                }

                return child;
              }
            })}
          </tr>
        </thead>
        <tbody className="table__body">
          {props.isLoading ? (
            <tr>
              <td colSpan={columns.length}>
                <Skeleton
                  count={5}
                  height={60}
                  style={{ marginTop: 10, zIndex: 0 }}
                />
              </td>
            </tr>
          ) : (
            data.map((row) => {
              return (
                <Fragment key={id + row[dataKey]}>
                  <TableRow
                    tableId={id}
                    columns={columns}
                    row={row}
                    checkable={props.checkable ?? false}
                    dataKey={dataKey}
                    selectedKeys={selectedKeys}
                    invertSelection={invertSelection}
                    onSelectionChange={props.onSelectionChange}
                    setSelectedKeys={setSelectedKeys}
                    onClick={props.onRowClick}
                    setDragging={props.setDragging}
                    onDrag={props.onDrag}
                    onDragEnd={props.onDragEnd}
                    draggable={!!props.onDragEnd}
                    watch={watch}
                    showColumnSelector={showColumnSelector}
                  />
                  {props.expandedRow && row[dataKey] === props.expandedRow && (
                    <tr className="table__row table__row--expanded">
                      <td colSpan={columns.length + 1}>
                        {props.renderExpandedRow?.(row)}
                      </td>
                    </tr>
                  )}
                </Fragment>
              );
            })
          )}
        </tbody>
        {props.withPagination && (
          <TableFooter
            currentPage={currentPage}
            pageCount={pageCount}
            setCurrentPage={setCurrentPage}
            onPageChange={(page) =>
              props.setFilters?.({ ...props.filters, page })
            }
          />
        )}
      </table>
    </>
  );
}
