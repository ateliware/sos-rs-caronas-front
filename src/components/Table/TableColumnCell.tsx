import { ColumnProps } from './TableColumn';

export default function TableColumnCell(props: ColumnProps & { row: any }) {
  const cellClasses = ['table__cell'];
  const cellContentClasses = ['table__cell__content'];
  if (props.ellipsis) cellClasses.push('table__cell--ellipsis');
  if (props.textAlign) cellClasses.push(`!text-${props.textAlign}`);
  if (props.verticalAlign) cellClasses.push(`!valign-${props.verticalAlign}`);
  if (props.cellAlign)
    cellContentClasses.push(`d-flex justify-${props.cellAlign}`);
  const row = props.row;
  const fromKey = props.fromKey;
  const content = props.render ? props.render(row[fromKey], row) : row[fromKey];

  return (
    <td className={cellClasses.join(' ')}>
      {(props.container && (
        <div className={cellContentClasses.join(' ')}>{content || '-'}</div>
      )) ||
        content ||
        '-'}
    </td>
  );
}
