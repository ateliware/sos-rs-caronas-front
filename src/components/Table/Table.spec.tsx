import { fireEvent, render, screen, waitFor } from '@testing-library/react';

import Table from './Table';
import TableColumn from './TableColumn';

describe('Table', () => {
  it('should render', () => {
    const { container } = render(<Table data={[]} />);
    expect(container).toMatchSnapshot();
  });

  it('should render with variants', () => {
    const { container } = render(
      <Table layout="fixed" className="custom" hoverable checkable data={[]} />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with data', async () => {
    const data = [
      {
        id: 1,
        name: 'John Doe',
        email: 'john@ateliware.com',
      },
    ];

    const { container } = await waitFor(() => {
      return render(
        <Table data={data} showColumnSelector={false}>
          <TableColumn fromKey="id" header="Id" />
          <TableColumn fromKey="name" header="Name" />
          <TableColumn fromKey="email" header="E-mail" />
        </Table>
      );
    });
    expect(container).toMatchSnapshot();

    const rows = container.querySelectorAll('tbody tr');
    expect(rows).toHaveLength(1);

    const columns = rows[0].querySelectorAll('td');
    expect(columns).toHaveLength(3);

    expect(columns[0]).toHaveTextContent('1');
    expect(columns[1]).toHaveTextContent('John Doe');
    expect(columns[2]).toHaveTextContent('john@ateliware.com');
  });

  it('should render with data and selected', () => {
    const data = [
      {
        id: 1,
        name: 'John Doe',
        email: 'john@ateliware.com',
      },
      {
        id: 2,
        name: 'Jane Doe',
        email: 'jane@ateliware.com',
      },
    ];

    const bulkAction = jest.fn();
    const { container } = render(
      <Table
        checkable
        data={data}
        selectedKeys={{ '1': true }}
        showColumnSelector={false}
        onAction={bulkAction}
      >
        <TableColumn fromKey="id" header="Id" />
        <TableColumn fromKey="name" header="Name" />
        <TableColumn
          fromKey="email"
          header="E-mail"
          verticalAlign="middle"
          textAlign="right"
          container
          ellipsis
        />
      </Table>
    );
    expect(container).toMatchSnapshot();

    const rows = container.querySelectorAll('tbody tr');
    expect(rows).toHaveLength(2);

    const columns = rows[0].querySelectorAll('td');
    expect(columns).toHaveLength(4);

    expect(columns[0]).toContainHTML('input');
    expect(columns[1]).toHaveTextContent('1');
    expect(columns[2]).toHaveTextContent('John Doe');
    expect(columns[3]).toHaveTextContent('john@ateliware.com');

    const bulkActionButton = container.querySelector(
      '.btn--table-action'
    ) as Element;
    expect(bulkActionButton).toBeInTheDocument();
    fireEvent.click(bulkActionButton);
    expect(bulkAction).toBeCalledWith({ '1': true }, false);

    const checkboxes = container.querySelectorAll('.form-check-input');
    expect(checkboxes).toHaveLength(3);

    let selected = container.querySelectorAll('.form-check-input:checked');
    expect(selected).toHaveLength(1);

    const selectAll = container.querySelector('.form-check-input') as Element;
    expect(selectAll).not.toBeChecked();
    fireEvent.click(selectAll);
    expect(selectAll).toBeChecked();

    selected = container.querySelectorAll('.form-check-input:checked');
    expect(selected).toHaveLength(3);

    fireEvent.click(bulkActionButton);
    expect(bulkAction).toBeCalledWith({}, true);

    const selectFirstBtn = container.querySelectorAll(
      '.form-check-input'
    )[1] as Element;
    expect(selectFirstBtn).toBeChecked();
    fireEvent.click(selectFirstBtn);
    expect(selectFirstBtn).not.toBeChecked();

    fireEvent.click(bulkActionButton);
    expect(bulkAction).toBeCalledWith({ '1': true }, true);
  });

  it('should render with search input', () => {
    const { container } = render(<Table data={[]} withSearchInput />);

    const tableFilter = container.querySelector('.table__filter');
    const input = screen.getByPlaceholderText('Pesquisar');

    expect(tableFilter).toBeInTheDocument();
    expect(input).toHaveAttribute('type', 'search');
  });

  it('should render expandable table', () => {
    const onRowClick = jest.fn();
    const row = { id: '1', name: 'John Doe' };
    render(
      <Table
        expandedRow="1"
        renderExpandedRow={() => 'Expanded'}
        onRowClick={onRowClick}
        data={[row]}
        showColumnSelector={false}
      >
        <TableColumn fromKey="id" header="Id" />
        <TableColumn fromKey="name" header="Name" />
      </Table>
    );

    const johnCell = screen.getByText('John Doe');
    const johnRow = johnCell.parentElement;
    const expandedCell = screen.getByText('Expanded');
    const expandedRow = expandedCell.parentElement;

    expect(expandedCell).toBeInTheDocument();
    expect(expandedCell).toHaveAttribute('colspan', '3');
    expect(expandedRow).toHaveClass('table__row table__row--expanded');

    fireEvent.click(johnRow as Element);
    expect(onRowClick).toBeCalledWith(row);
  });

  it('should render draggable table', async () => {
    const onDrag = jest.fn();
    const data = [
      {
        id: 1,
        name: 'John Doe',
        email: 'email@mail.com',
      },
      {
        id: 2,
        name: 'Jane Doe',
        email: 'email2@mail.com',
      },
    ];

    const { container } = render(
      <Table
        data={data}
        onDrag={onDrag}
        setDragging={jest.fn()}
        onDragEnd={jest.fn()}
      >
        <TableColumn fromKey="id" header="Id" />
        <TableColumn fromKey="name" header="Name" />
        <TableColumn fromKey="email" header="E-mail" />
      </Table>
    );

    await waitFor(() => {
      return expect(screen.getAllByText('drag_indicator')).toHaveLength(2);
    });

    const draggableRow = container.querySelector('.table__row') as Element;
    const dropZone = container.querySelectorAll('.table__row')[1] as Element;

    fireEvent.mouseDown(draggableRow, { which: 1, button: 0 });
    fireEvent.dragOver(dropZone);
    fireEvent.drop(dropZone);
    fireEvent.mouseUp(dropZone, { which: 1, button: 0 });

    expect(onDrag).toBeCalledWith(1);
  });

  it('should render with loading', () => {
    const { container } = render(<Table isLoading data={[]}></Table>);

    const rows = container.querySelectorAll('.react-loading-skeleton');
    expect(rows).toHaveLength(5);
  });

  it('should render footer with pagination', async () => {
    const setFilters = jest.fn();

    const { container } = render(
      <Table
        filters={{}}
        setFilters={setFilters}
        withPagination
        totalRecords={100}
        isLoading
        data={[]}
      ></Table>
    );

    const tfoot = container.querySelector('tfoot');
    const paginationContainer = container.querySelector(
      '.pagination-container'
    );

    expect(tfoot).toBeInTheDocument();
    expect(paginationContainer).toBeInTheDocument();
  });
});
