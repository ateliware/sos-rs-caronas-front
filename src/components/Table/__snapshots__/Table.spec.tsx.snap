// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render 1`] = `
<div>
  <div
    class="table__filter col-md-12"
  >
    <div
      class="d-flex justify-between"
    >
      <div
        class="popover-wrapper"
      >
        <div
          class="popover-trigger"
        >
          <div
            class="d-flex align-items-center justify-center pl-s-300 pr-s-100"
          >
            <div
              class="showColumnSelector d-flex align-items-center"
              title="Exibir colunas"
            >
              <span
                class="material-symbols-rounded"
              >
                visibility
              </span>
              <span
                class="ml-s-100"
              >
                Colunas
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <table
    class="table"
    id=":r0:"
  >
    <thead
      class="table__head"
    >
      <tr
        class="table__row"
      />
    </thead>
    <tbody
      class="table__body"
    />
  </table>
</div>
`;

exports[`Table should render with data 1`] = `
<div>
  <div
    class="table__filter col-md-12"
  >
    <div
      class="d-flex justify-between"
    />
  </div>
  <table
    class="table"
    id=":r2:"
  >
    <thead
      class="table__head"
    >
      <tr
        class="table__row"
      >
        <th
          class="table__cell"
        >
          <div
            class="table__cell__content"
          >
            Id
          </div>
        </th>
        <th
          class="table__cell"
        >
          <div
            class="table__cell__content"
          >
            Name
          </div>
        </th>
        <th
          class="table__cell"
        >
          <div
            class="table__cell__content"
          >
            E-mail
          </div>
        </th>
      </tr>
    </thead>
    <tbody
      class="table__body"
    >
      <tr
        class="table__row"
        draggable="false"
      >
        <td
          class="table__cell"
        >
          1
        </td>
        <td
          class="table__cell"
        >
          John Doe
        </td>
        <td
          class="table__cell"
        >
          john@ateliware.com
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Table should render with data and selected 1`] = `
<div>
  <div
    class="table__filter col-md-12"
  >
    <div
      class="d-flex justify-between"
    />
  </div>
  <table
    class="table"
    id=":r3:"
  >
    <thead
      class="table__head"
    >
      <tr
        class="table__row"
      >
        <th
          class="table__cell table__cell--checkbox"
        >
          <div
            class="table__cell__content"
          >
            <div
              class="form-check"
            >
              <input
                class="form-check-input"
                id="checkbox"
                type="checkbox"
              />
            </div>
            <button
              class="btn btn--icon-square btn--table-action ml-s-50"
            >
              <div
                class="btn__body"
              >
                <span
                  class="material-symbols-outlined btn__body__icon"
                >
                  arrow_drop_down
                </span>
              </div>
            </button>
          </div>
        </th>
        <th
          class="table__cell"
        >
          <div
            class="table__cell__content"
          >
            Id
          </div>
        </th>
        <th
          class="table__cell"
        >
          <div
            class="table__cell__content"
          >
            Name
          </div>
        </th>
        <th
          class="table__cell"
        >
          <div
            class="table__cell__content"
          >
            E-mail
          </div>
        </th>
      </tr>
    </thead>
    <tbody
      class="table__body"
    >
      <tr
        class="table__row"
        draggable="false"
      >
        <td
          class="table__cell table__cell--checkbox"
        >
          <div
            class="table__cell__content"
          >
            <div
              class="form-check"
            >
              <input
                checked=""
                class="form-check-input"
                id="checkbox"
                type="checkbox"
              />
            </div>
          </div>
        </td>
        <td
          class="table__cell"
        >
          1
        </td>
        <td
          class="table__cell"
        >
          John Doe
        </td>
        <td
          class="table__cell table__cell--ellipsis !text-right !valign-middle"
        >
          <div
            class="table__cell__content"
          >
            john@ateliware.com
          </div>
        </td>
      </tr>
      <tr
        class="table__row"
        draggable="false"
      >
        <td
          class="table__cell table__cell--checkbox"
        >
          <div
            class="table__cell__content"
          >
            <div
              class="form-check"
            >
              <input
                class="form-check-input"
                id="checkbox"
                type="checkbox"
              />
            </div>
          </div>
        </td>
        <td
          class="table__cell"
        >
          2
        </td>
        <td
          class="table__cell"
        >
          Jane Doe
        </td>
        <td
          class="table__cell table__cell--ellipsis !text-right !valign-middle"
        >
          <div
            class="table__cell__content"
          >
            jane@ateliware.com
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Table should render with variants 1`] = `
<div>
  <div
    class="table__filter col-md-12"
  >
    <div
      class="d-flex justify-between"
    >
      <div
        class="popover-wrapper"
      >
        <div
          class="popover-trigger"
        >
          <div
            class="d-flex align-items-center justify-center pl-s-300 pr-s-100"
          >
            <div
              class="showColumnSelector d-flex align-items-center"
              title="Exibir colunas"
            >
              <span
                class="material-symbols-rounded"
              >
                visibility
              </span>
              <span
                class="ml-s-100"
              >
                Colunas
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <table
    class="table table--fixed table--hoverable custom"
    id=":r1:"
  >
    <thead
      class="table__head"
    >
      <tr
        class="table__row"
      >
        <th
          class="table__cell table__cell--checkbox"
        >
          <div
            class="table__cell__content"
          >
            <div
              class="form-check"
            >
              <input
                class="form-check-input"
                id="checkbox"
                type="checkbox"
              />
            </div>
          </div>
        </th>
      </tr>
    </thead>
    <tbody
      class="table__body"
    />
  </table>
</div>
`;
